analyzere/utils.py

```diff
@@ -92,3 +92,17 @@ def vectorize(values):
     if isinstance(values, list):
         return ','.join(str(v) for v in values)
     return values
+
+
+def vectorize_range(values):
+    """
+    Takes a value or list of tuples and returns a single result,
+    tuples are joined by "," if necessary, elements in tuple are joined by '_'
+    """
+    if isinstance(values, tuple):
+        return '_'.join(str(i) for i in values)
+    elif isinstance(values, list):
+        if all([isinstance(item, tuple) for item in values]):
+            return ','.join('_'.join(str(i) for i in v) for v in values)
+        raise Exception('values must be a tuple or list of tuples')
```

Comment:
This error message isn't quite correct. 

It should read "values must be tuples" if the error message is in regards to the condition when the input is a list. OR the final "else" expression (the condition where the input is not a tuple or list of tuples) should also throw.

Reply:
yeah, this function allow a single value(not tuple), or a tuple,  or a list of tuples. I don't want the error message says `"values must be tuples"`, because we do allow a tuple or a single value.
Maybe this one, `"items in the list must be tuples"`?